incremental-font-transfer/src/bin/ift_extend.rs

```diff
@@ -82,41 +94,78 @@ fn main() {
 
     let mut patch_data: HashMap<String, UriStatus> = Default::default();
     let mut it_count = 0;
+
+    // For this a roundtrip is defined as an iteration of the below loop where at least
+    // one new patch URI is needed.
+    let mut round_trip_count = 0;
+    let mut fetch_count = 0;
     loop {
         it_count += 1;
         println!(">> Iteration {}", it_count);
         let font = FontRef::new(&font_bytes).expect("Input font parsing failed");
-        let next_patches = PatchGroup::select_next_patches(font, &subset_definition)
+        let next_patches = PatchGroup::select_next_patches(font, &patch_data, &subset_definition)
             .expect("Patch selection failed");
         if !next_patches.has_uris() {
-            println!("  No outstanding patches, all done.");
+            println!("    No outstanding patches, all done.");
             break;
         }
 
-        println!("  Selected URIs:");
+        let mut fetched = false;
         for uri in next_patches.uris() {
-            println!("    fetching {}", uri);
-            let uri_path = args.font.parent().unwrap().join(uri);
-            let patch_bytes = std::fs::read(uri_path.clone()).unwrap_or_else(|e| {
-                panic!(
-                    "Unable to read patch file ({}): {:?}",
-                    uri_path.display(),
-                    e
-                )
+            patch_data.entry(uri.to_string()).or_insert_with_key(|key| {
+                let uri_path = args.font.parent().unwrap().join(uri);
+                println!("    Fetching {}", key);
+                fetched = true;
+                fetch_count += 1;
+                if let Some(max_fetch_count) = args.max_fetches {
+                    if fetch_count > max_fetch_count {
+                        panic!(
+                            "Maximum number of fetches ({} > {}) exceeded.",
+                            fetch_count, max_fetch_count
+                        );
+                    }
+                }
+
+                let patch_bytes = std::fs::read(uri_path.clone()).unwrap_or_else(|e| {
+                    panic!(
+                        "Unable to read patch file ({}): {:?}",
+                        uri_path.display(),
+                        e
+                    )
+                });
```

Comment:
appreciate this isn't new code but would note that if you want to panic it would be more idiomatic to `match` instead of using `unwrap_or_else`, which is intended to be used to compute a value.